### ISSUE TITLE
FNX-4793 ⁃ Inconsistency & layout issues in Custom tab menus

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/fenix#1092]
## Steps to reproduce
1. Open a custom tab in Slack.
1. Open the menu.
    1. Expected behavior
The app has a customized custom tab menu.
    1. Actual behavior
On some devices: i.e. Huawei Honor 8 (Android 7.0), Nexus 5 (Android 6.0.1)

* the customized menu on Slack has the items in reversed order. see screenshot:
![Screenshot](https://user-images.githubusercontent.com/4037709/54674445-1d4cc180-4b05-11e9-9ebe-63b69587ed00.png)UNITO-UNDERSCORE!20190320-105037!
* On Samsung S8 (Android 9), the Slack custom tab menu is not customized, the app's full menu is displayed: 
    1. Device information
* Android device: see above
* Fenix version: 3/19 build

**Comment by bsurd** [comment]
According to #203 the custom menu should be above the Fenix menu, this is still not the case for the affected devices.

If logged into the Slack app and opening a custom tab the custom menu is still not displayed. Eg. opened a wikipedia link, custom slack menu is not available.

I've also logged a new issue for Samsung devices #1767.

**Comment by bsurd** [comment]
@sblatz any input here?

**Comment by sblatz** [comment]
This is really strange. The custom menu is also not showing for me in Chrome, Firefox, Firefox Focus, or Reference Browser on my Galaxy S9. I'm not sure this is something we actually are doing incorrectly, and may just be an incompatibility based on device type.

@bsurd can you list which devices are utilizing the custom menu and which aren't? Also, are they all at least showing the "shortened" menu (the one that says "Powered by Fenix")?

**Comment by bsurd** [comment]
Yes, they are showing the shortened menu "Powered by Fenix", I've used the same devices as mentioned in the initial report the Honor, 8 Nexus 5 and Galaxy S8, as well as a HTC 10 and Note 8 beside those.

Logged the new issue while testing for Samsung devices as they seem to not even open a custom tab.

Please let me know if there is any other information I can help with.

**Comment by bsurd** [comment]
The inconsistency is noticeable in the Samsung devices, after logging in and opening a custom tab only the shortened menu is displayed, while on other devices the slack custom menu is also displayed alongside it.

**Comment by sblatz** [comment]
@vesta0 how do we want to proceed here? I believe this bug is not our fault (as I can reproduce in all the browsers on my phone).

**Comment by sblatz** [comment]
I verified this on the devices below:

On Samsung Galaxy S9: No browsers show “Share in Slack…”
On Pixel 3: All browsers show “Share in Slack…”

This seems to not be a Fenix issue, and thus I'm closing per @vesta0's recommendation.

**Comment by bsurd** [comment]
Reopening as part of the initial issue still remains, the items are still displayed in reverse order compared to the specs in #203.

> ### Actual behavior
> * the customized menu on Slack has the items in reversed order. see screenshot:

![img](https://user-images.githubusercontent.com/17178945/56415356-2cef2f80-6296-11e9-8d96-01cf240ecfe8.jpg)

**Comment by bifleming** [comment]
@sblatz @vesta0 is this still blocked? Note that it was re-opened.

**Comment by sblatz** [comment]
I submitted an a-c patch for this [here](https://github.com/mozilla-mobile/android-components/pull/2809). This will once again be blocked on it landing.

**Comment by sblatz** [comment]
https://github.com/mozilla-mobile/android-components/pull/2809 was merged today. Should be in nightly tonight.

**Comment by kbrosnan** [comment]
@sblatz still does not match spec in comment 7.

![Screenshot_20190425-183153](https://user-images.githubusercontent.com/250273/56777905-9edce280-6788-11e9-90d7-fb792834359b.png)

**Comment by sv-ohorvath** [comment]
I think this is finally fixed :)
Customized/standard menus are looking good:
![Screenshot_20190430-160204](https://user-images.githubusercontent.com/4037709/56963442-9fe38a80-6b61-11e9-8eab-9f2e54a099a6.png)
